src/checkers/inference/model/ComparisonConstraint.java

```diff
@@ -0,0 +1,140 @@
+package checkers.inference.model;
+
+import java.util.Arrays;
+
+import org.checkerframework.framework.type.QualifierHierarchy;
+import org.checkerframework.javacutil.BugInCF;
+
+import com.sun.source.tree.Tree.Kind;
+
+/**
+ * Represents a constraint that two slots must be comparable.
+ *
+ */
+public class ComparisonConstraint extends Constraint {
+
+    private final ComparisonOperationKind operation;
+    private final Slot left;
+    private final Slot right;
+    private final ComparisonVariableSlot result;
+    
+    public enum ComparisonOperationKind {
+        EQUAL_TO("=="),
+        NOT_EQUAL_TO("!="),
+        GREATER_THAN(">"),
+        GREATER_THAN_EQUAL(">="),
+        LESS_THAN("<"),
+        LESS_THAN_EQUAL("<=");
+
+        // stores the symbol of the operation
+        private final String opSymbol;
+
+        private ComparisonOperationKind(String opSymbol) {
+            this.opSymbol = opSymbol;
+        }
+
+        public static ComparisonOperationKind fromTreeKind(Kind kind) {
+            switch (kind) {
+                case EQUAL_TO:
+                	return EQUAL_TO;
+                case NOT_EQUAL_TO:
+                	return NOT_EQUAL_TO;
+                case GREATER_THAN:
+                	return GREATER_THAN;
+                case GREATER_THAN_EQUAL:
+                	return GREATER_THAN_EQUAL;
+                case LESS_THAN:
+                	return LESS_THAN;
+                case LESS_THAN_EQUAL:
+                	return LESS_THAN_EQUAL;
+                default:
+                	throw new BugInCF("There are no defined ComparableOperationKind "
+                            + "for the given com.sun.source.tree.Tree.Kind: " + kind);
+            }
+        }
+
+        public String getSymbol() {
+            return opSymbol;
+        }
+    }
+
+    private ComparisonConstraint(ComparisonOperationKind operation, Slot left, Slot right,
+            ComparisonVariableSlot result, AnnotationLocation location) {
+        super(Arrays.asList(left, right, result), location);
+        this.left = left;
+        this.right = right;
+        this.operation = operation;
+        this.result = result;
+    }
+
+    private ComparisonConstraint(ComparisonOperationKind operation, Slot left, Slot right, 
```

Comment:
This constructor is never used. Shall we remove it?

Every ComparisonConstraint should have a location